src/platform/datapath_raw_socket.c

```diff
@@ -379,11 +445,18 @@ CxPlatFramingWriteHeaders(
         IPv4->TimeToLive = IP_DEFAULT_HOP_LIMIT;
         IPv4->Protocol = IPPROTO_UDP;
         IPv4->HeaderChecksum = 0;
-        CxPlatCopyMemory(IPv4->Destination, &Route->RemoteAddress.Ipv4.sin_addr, sizeof(Route->RemoteAddress.Ipv4.sin_addr));
         CxPlatCopyMemory(IPv4->Source, &Route->LocalAddress.Ipv4.sin_addr, sizeof(Route->LocalAddress.Ipv4.sin_addr));
+        CxPlatCopyMemory(IPv4->Destination, &Route->RemoteAddress.Ipv4.sin_addr, sizeof(Route->RemoteAddress.Ipv4.sin_addr));
+        IPv4->HeaderChecksum = SkipNetworkLayerXsum ? 0 : ~CxPlatFramingChecksum((uint8_t*)IPv4, sizeof(IPV4_HEADER), 0);
         EthType = ETHERNET_TYPE_IPV4;
         Ethernet = (ETHERNET_HEADER*)(((uint8_t*)IPv4) - sizeof(ETHERNET_HEADER));
         IpHeaderLen = sizeof(IPV4_HEADER);
+        UDP->Checksum =
```

Comment:
Both here and below, since you've already written 0 to `Checksum` can you just wrap this code in an `if (!SkipTransportLayerXsum) {...}`?

Reply:
Sure, I initially thought 2 ternary operations would be faster than a branch but since there's a function call, ternary operation won't be optimized anyway.

Reply:
I bet it won't make any difference in assembly. I can do it for readability.